src/nl/rubensten/texifyidea/inspections/TexifyRegexInspection.kt

```diff
@@ -172,31 +239,72 @@ abstract class TexifyRegexInspection(
         return mathMode == element.inMathContext() && checkContext(element)
     }
 
+    override fun runForWholeFile(): Boolean {
+        // We assume the quickfix of this inspection replaces text (like <<) by
+        // content with a different length (like \ll), so we have to make
+        // sure it is run for the whole file at once.
+        // It can be disabled by overriding this method.
+        return true
+    }
+
     /**
      * Replaces all text in the replacementRange by the correct replacement.
+     *
+     * @return The total increase in document length, e.g. if << is replaced by \ll and \usepackage{amsmath} is added then the total increase is 3 + 20 - 2.
      */
-    open fun applyFix(project: Project, descriptor: ProblemDescriptor, replacementRange: IntRange, replacement: String, groups: List<String>) {
+    open fun applyFix(project: Project, descriptor: ProblemDescriptor, replacementRange: IntRange, replacement: String, groups: List<String>): Int {
         val file = descriptor.psiElement as PsiFile
-        val document = file.document() ?: return
+        val document = file.document() ?: return 0
 
         document.replaceString(replacementRange.start, replacementRange.endInclusive, replacement)
+
+        return replacement.length - (replacementRange.endInclusive - replacementRange.start)
+    }
+
+    /**
+     * Replaces all text for all replacementRanges by the correct replacements.
+     *
+     * We assume the quickfix of an inspection performs replacements which could
+     * have a different length than the original text, the inspection
+     * should override [runForWholeFile] to explicitly refute this assumption.
+     *
+     * @param groups Regex groups as matched by the regex matcher.
+     */
+    open fun applyFixes(project: Project, descriptor: ProblemDescriptor, replacementRanges: List<IntRange>, replacements: List<String>, groups: List<List<String>>) {
+        assert(replacementRanges.size == replacements.size)
+
+        // Remember how much a replacement changed the locations of the fixes still to be applied.
+        // This is cumulative, but may be negative.
+        var accumulatedDisplacement = 0
+
+        // Loop over all fixes manually, in order to fix the regex locations
+        for (i in 0 until replacements.size) {
+            val replacementRange = replacementRanges[i]
+            val replacement = replacements[i]
+
+            val newRange = IntRange(replacementRange.start + accumulatedDisplacement, replacementRange.endInclusive + accumulatedDisplacement)
```

Comment:
I think the approach is good, although the implementation assumes the `replacementRanges` are increasing and non-overlapping. You can check for this beforehand (with require), but this is relatively expensive. I think it is best that this requirement is explicitly documented in the doc comment.

Since `Matcher.find()` indeed only matches increasingly in the search text and only returns non-overlapping matches, the requirements are satisfied. Since this behavior of `Matcher.find()` is not immediately obvious, mention this in a comment at `TexifyRegexInspection.kt:138`.